src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -2482,7 +2482,8 @@ bool LibraryCallKit::inline_vector_convert() {
   // through a vector or predicate register.
   if (is_mask &&
       ((src_type->isa_vectmask() == NULL && dst_type->isa_vectmask()) ||
-       (dst_type->isa_vectmask() == NULL && src_type->isa_vectmask()))) {
+       (dst_type->isa_vectmask() == NULL && src_type->isa_vectmask()) ||
+       num_elem_from != num_elem_to)) {
```

Comment:
Yes, it is. Just want to double check in the compiler side, or it maybe confusing if someone is not familiar with the vector api java codes.

Reply:
I see, may be we can just add a comment instead of injecting code.

Reply:
Yeah, agree to you. Or maybe as assert under `if(num_elem_from < num_elem_to)` and `if(num_elem_from > num_elem_to)` ?  I'm sorry that I didn't find a better place to add the comment. Any suggestions? For me, it's better to add the comment inside the `if(is_cast)` block since it has different paths based on the `num_elem_from` and `num_elem_to`.

Reply:
Something like: `assert(!is_mask, "mask cast needs the same elem num ")`

Reply:
Hi, I'd like the add the assertion in the unify patch instead of this. So I will revert this change here. Thanks!

Reply:
Yes, assert (!is_mask || (is_mask && num_elem_from == num_elem_to),  "").

Reply:
Looks good. Thanks!

Reply:
The new commit reverted this changes. I will add the assertion in https://github.com/openjdk/jdk/pull/10192.